requests_oauthlib/oauth2_session.py

```diff
@@ -197,7 +197,8 @@ def fetch_token(self, token_url, code=None, authorization_response=None,
                 redirect_uri=self.redirect_uri, username=username,
                 password=password, **kwargs)
 
-        auth = auth or requests.auth.HTTPBasicAuth(username, password)
+        if not auth and (username):
+            auth = auth or requests.auth.HTTPBasicAuth(username, password)
```

Comment:
Let's tighten this up a little bit if you don't mind. =)

Firstly, we can remove the `auth or` portion of this line: we already know `auth` is false, that's how we got here, so no need to check it again.

Secondly, let's add a line inside this block that asserts that `password` is truthy.

Thirdly and finally, let's remove the parentheses around `username` in the conditional: they're not needed. =)